Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM alpine:3.14 AS sslocal-builder
+
+ENV PATH=/root/.cargo/bin:$PATH
+
+RUN apk add build-base curl
+
+RUN curl https://sh.rustup.rs -o rustup-init.sh && sh rustup-init.sh -y --default-toolchain nightly
+
+WORKDIR /root/shadowsocks-rust
+
+ADD . .
+
+RUN cargo build --release --bin sslocal
```

Comment:
Just run `cargo build --release`. It starts building all binaries at the same time when all dependencies are built, so it can take advantage of multiple CPU threads. And it also gives you `ssurl` and `ssmanager`.

Reply:
Yes, your solution can be effective in reducing time when building all binaries. But this Dockerfile is only for users building on their own, and they often only need one platform and one binary.

My plan is to have the CI use a separate Dockerfile, because performance is extremely poor when the CI uses Docker BuildX with QEMU to build ARM images. For example, building an amd64 image takes only 7 to 10 minutes, while building an arm64 image takes over 40 minutes, which is unacceptable.

I will later provide [cross](https://github.com/rust-embedded/cross) based builds for CI to solve this problem. After that I will implement your solution, compare them and choose the best one, thank you for your advice.